chasm/engine.go

```diff
@@ -42,8 +42,7 @@ type Engine interface {
 	PollComponent(
 		context.Context,
 		ComponentRef,
-		func(Context, Component) (any, bool, error),
-		func(MutableContext, Component, any) error,
+		func(Context, Component) (bool, error),
```

Comment:
I've removed `operationFn` from `PollComponent` since we don't need poll-with-mutation yet.